### PR TITLE
Disk eviction node e2e test

### PR DESCRIPTION
**Pull request by ronnielai**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30721)

Please review d202816cfb4ae4dc38d0fe4f45cd11cc81e0e178 only.
#29800

### CODE REVIEW

File: test/e2e_node/disk_eviction_test.go [hunk 1]
```diff
@@ -142,18 +136,30 @@ var _ = framework.KubeDescribe("Kubelet Eviction Manager [Flaky] [Serial] [Disru
...
+							}
 						}
-
 						evictionOccurred = true
+						return fmt.Errorf("waiting for node disk pressure condition to be set")
+					}
+
+					// The node should have disk pressure condition after the pods are evicted.
+					if !nodeDiskPressureCondition {
+						if !nodeHasDiskPressure(f.Client) {
```
Comment (Random-Liu) [inline test/e2e_node/disk_eviction_test.go]: Why moving the disk pressure check after the pod eviction check? Is it possible that node is relieved quickly after pod gets evicted.

Comment (ronnielai) [inline test/e2e_node/disk_eviction_test.go]: The rearrangement is to make the code cleaner (I hope)

The node's disk pressure can only be relieved after 1 minute the disk pressure has been set per the configuration.

**Comment by Random-Liu** [comment]
LGTM.

**Comment by ronnielai** [comment]
Reapplied LGTM after rebasing

**Pull request merged**